File: ios-oc/VoiceAgent/Chat/ConnectionStartView.m
//
//  ConnectionStartView.m
//  VoiceAgent
//
//  Created by qinhui on 2025/11/17.
//

#import "ConnectionStartView.h"
#import <Masonry/Masonry.h>

static inline UIColor *VAHexColor(NSUInteger hexValue, CGFloat alpha) {
    return [UIColor colorWithRed:((hexValue >> 16) & 0xFF) / 255.0
                           green:((hexValue >> 8) & 0xFF) / 255.0
                            blue:(hexValue & 0xFF) / 255.0
                           alpha:alpha];
}

@interface ConnectionStartView ()

@property (nonatomic, strong, readwrite) UIButton *startButton;

@end

@implementation ConnectionStartView

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self setupUI];
        [self setupConstraints];
    }
    return self;
}

- (instancetype)init {
    return [self initWithFrame:CGRectZero];
}

- (void)setupUI {
    self.backgroundColor = [UIColor clearColor];
    
    self.startButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.startButton setTitle:@"Start Agent" forState:UIControlStateNormal];
    [self.startButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [self.startButton setTitleColor:[[UIColor whiteColor] colorWithAlphaComponent:0.5] 
                            forState:UIControlStateDisabled];
    self.startButton.titleLabel.font = [UIFont systemFontOfSize:14 weight:UIFontWeightMedium];
    self.startButton.backgroundColor = VAHexColor(0x2563EB, 1.0);
    self.startButton.layer.cornerRadius = 8;
    self.startButton.enabled = YES;
    [self addSubview:self.startButton];
}

- (void)setupConstraints {
    [self.startButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.equalTo(self).inset(20);
        make.bottom.equalTo(self.mas_safeAreaLayoutGuideBottom).offset(-20);
        make.height.mas_equalTo(48);
    }];
}

- (void)updateButtonState:(BOOL)isEnabled {
    self.startButton.enabled = isEnabled;
    self.startButton.backgroundColor = isEnabled ? 
        VAHexColor(0x2563EB, 1.0) : 
        VAHexColor(0x334155, 1.0);
}

@end

File: ios-swift/VoiceAgent/Chat/ConnectionStartView.swift
//
//  ConfigBackgroundView.swift
//  VoiceAgent
//
//  Created by qinhui on 2025/11/17.
//

import UIKit
import SnapKit

class ConnectionStartView: UIView {
    // MARK: - UI Components
    let startButton = UIButton(type: .system)

    // MARK: - Initialization
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI()
        setupConstraints()
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    // MARK: - UI Setup
    private func setupUI() {
        backgroundColor = .clear

        startButton.setTitle("Start Agent", for: .normal)
        startButton.setTitleColor(.white, for: .normal)
        startButton.setTitleColor(AppColors.btnDisabledText, for: .disabled)
        startButton.titleLabel?.font = .systemFont(ofSize: 14, weight: .medium)
        startButton.backgroundColor = AppColors.btnStartBg
        startButton.layer.cornerRadius = 8
        startButton.isEnabled = true
        addSubview(startButton)
    }

    private func setupConstraints() {
        startButton.snp.makeConstraints { make in
            make.left.right.equalToSuperview().inset(20)
            make.bottom.equalTo(safeAreaLayoutGuide).offset(-20)
            make.height.equalTo(48)
        }
    }

    // MARK: - Public Methods
    func updateButtonState(isEnabled: Bool) {
        startButton.isEnabled = isEnabled
        startButton.backgroundColor = isEnabled ? AppColors.btnStartBg : AppColors.btnDisabledBg
    }
}
